File: 884. Uncommon Words from Two Sentences/Solution.py
class Solution:
    def store_string(self, count, s):
        words = s.split(" ")
        for word in words:
            if word in count:
                count[word] += 1
            else:
                count[word] = 1
                
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        count = {}
        self.store_string(count, s1)
        self.store_string(count, s2)
        
        ans = [word for word, freq in count.items() if freq == 1]
        return ans
        

File: 884. Uncommon Words from Two Sentences/Solution.java
class Solution {
    
    private void storeString(Map<String, Integer> count, String s) {
        String[] words = s.split(" "); 
        for (String word : words) {
            count.put(word, count.getOrDefault(word, 0) + 1);
        }
    }

    public String[] uncommonFromSentences(String s1, String s2) {
        Map<String, Integer> count = new HashMap<>();
        
        storeString(count, s1);
        storeString(count, s2);

        List<String> uncommonWords = new ArrayList<>();
        for (Map.Entry<String, Integer> entry : count.entrySet()) {
            if (entry.getValue() == 1) {
                uncommonWords.add(entry.getKey());
            }
        }

        return uncommonWords.toArray(new String[0]);
    }
}